app/src/main/java/org/mozilla/fenix/settings/logins/SavedLoginsView.kt

```diff
@@ -73,29 +79,61 @@ class SavedLoginsView(
 
 /**
  * Interactor for the saved logins screen
+ *
+ * @param savedLoginsController [SavedLoginsController] which will be delegated for all users interactions.
  */
 class SavedLoginsInteractor(
-    private val savedLoginsController: SavedLoginsController,
-    private val itemClicked: (SavedLogin) -> Unit,
-    private val learnMore: () -> Unit
+    private val savedLoginsController: SavedLoginsController
 ) {
-    fun itemClicked(item: SavedLogin) {
-        itemClicked.invoke(item)
+    fun onItemClicked(item: SavedLogin) {
+        savedLoginsController.handleItemClicked(item)
     }
-    fun onLearnMore() {
-        learnMore.invoke()
+    fun onLearnMoreClicked() {
+        savedLoginsController.handleLearnMoreClicked()
     }
-    fun sort(sortingStrategy: SortingStrategy) {
+    fun onSortingStrategyChanged(sortingStrategy: SortingStrategy) {
         savedLoginsController.handleSort(sortingStrategy)
     }
 }
 
 /**
  * Controller for the saved logins screen
+ *
+ * @param store Store used to hold in-memory collection state.
+ * @param navController NavController manages app navigation within a NavHost.
+ * @param browserNavigator Controller allowing browser navigation to any Uri.
+ * @param settings SharedPreferences wrapper for easier usage.
+ * @param metrics Controller that handles telemetry events.
```

Comment:
Same thing here, I do not want the metrics component passed since it doesn't have anything to do with the logic of the controller.

Reply:
As above, the way I see it, the controller is the only one that knows that a certain event needs to be logged or not. It handles the business logic.
So everything needed to do it's job needs to preferably be injected in the constructor.